app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -372,4 +374,29 @@ public Task<Integer> calcPinCost(@NonNull Location location) {
         return functions.getHttpsCallable("calcPinCost").call(data)
                 .continueWith(task -> (Integer) task.getResult().getData());
     }
+
+    public void postComment(Comment comment, String pid) {
+        db.collection("pins")
+                .document(pid)
+                .collection("comments")
+                .add(comment.serialize())
+                .addOnSuccessListener(t -> Log.d(TAG, String.format("Comment on pin %s successfully posted.", pid)))
+                .addOnFailureListener(e -> Log.w(TAG, "Error posting comment", e));
+    }
```

Comment:
We should probably return the upload task so we can handle that in the client. Like either display toast saying upload failed or add the comment to the list of comments only if it succeeds.